robot/pmm/tests/browser/ServiceDeliveries/new_service_delivery.robot

```diff
@@ -18,67 +18,74 @@ Setup Test Data
+    Page Should Not Contain                ${service_delivery_name}
+    Verify Page Contains Related List      Files
+    Save Current Record ID For Deletion     ${ns}ServiceDelivery__c
 
 Create a Service Delivery via UI with Auto Name Override
-    [Documentation]                        This test creates Service Delivery record and verifies that the Service Delivery record
-    ...                                    has all the values from the form
+    [Documentation]                        This test creates Service Delivery record with auto name override selected and verifies
+    ...                                    user entered name is saved as the service delivery name
     Go To Page                             Listing                                ${ns}ServiceDelivery__c
     Click Object Button                    New
-    Load Page Object                       NewServiceDelivery                     ${ns}ServiceDelivery__c
-    verify current page title              New Service Delivery
-    Populate modal Form                    Service Delivery Name=${service_delivery_name}
-    ...                                    Service=&{service}[Name]
-    ...                                    Client=&{contact}[FirstName] &{contact}[LastName]
-    ...                                    Program Engagement=&{program_engagement}[Name]
+    Wait For Modal                         New                                    Service Delivery
+    Populate Modal Form                    Service Delivery Name=${service_delivery_name}
+    ...                                    Service=${service}[Name]
+    ...                                    Client=${contact}[FirstName] ${contact}[LastName]
+    ...                                    Program Engagement=${program_engagement}[Name]
     ...                                    Delivery Date=Today
     ...                                    Quantity=${quantity}
     ...                                    Auto-name Override=checked
-    click modal button                     Save
+    Click Modal Button                     Save
     Wait Until Modal Is Closed
-    verify page header                     Service Delivery
-    verify details                         Service Delivery Name                   contains             ${service_delivery_name}
-    verify page contains related list      Files
-    ${service_delivery_id} =               Save Current Record ID For Deletion     ${ns}ServiceDelivery__c
-    ${service_id} =                        Save Current Record ID For Deletion     ${ns}Service__c
-    ${program_engagement_id} =             Save Current Record ID For Deletion     ${ns}ProgramEngagement__c
\ No newline at end of file
+    Current Page Should Be                 Details                                 ServiceDelivery__c
+   # Verify Page Header                     Service Delivery
```

Comment:
remove this commented out line

Reply:
Removed and also updated the is_current_page keyword